src/sqlfluff/core/rules/base.py

```diff
@@ -499,6 +499,7 @@ class RuleMetaclass(type):
         flags=re.MULTILINE,
     )
     _valid_classname_regex = regex.compile(r"Rule_?([A-Z]{1}[a-zA-Z]+)?_([A-Z0-9]{4})")
+    _valid_rule_name_regex = regex.compile(r"[a-z\.\_]+")
```

Comment:
thoughts on `a-z[a-z\.\_]+`?

Reply:
yeah, that would be better... I'll update it.